### DISCUSSION TITLE
cc2500 frequency tuning

### DISCUSSION THREAD

**Discussion by mha1** [pascallanger/DIY-Multiprotocol-TX-Module#684] [Q&A]
@pascallanger 

Hi Pascal,

I have a question about CC2500 manual frequency tuning. As far as I know this is a non-native feature for radio systems using the CC2500, i.e. FrSky, Graupner HoTT and probaly other CC2500 radio systems don't offer this on their transmitters (and receivers). This is why I wonder why you chose to offer this.

If I understand the CC2500 data sheet correctly the CC2500 estimates the offset between transmitter and receiver frequencies and writes this value to the FREQEST ststus register (section 14.1). Section 14.1 then states "Writing the value from FREQEST into
FSCTRL0.FREQOFF the frequency synthesizer is automatically adjusted according to the estimated frequency offset.". I would assume this is the way all the OEM receivers and transmitters do this. 

In `static void __attribute__((unused)) HOTT_tune_freq()` you choose to use the manually generated frequency offset value `CC2500_WriteReg(CC2500_0C_FSCTRL0, option);`  

Did you encounter problems with the offset estimate or don't trust it?

Thanks for your insight.

Have a great day
Michael

**Comment by mha1** [comment]
Anyone?

**Reply by pascallanger** [reply]
You are assuming that every CC2500 protocol is connected to a receiver sending back telemetry which is not the case. So yes while you could use this information for some protocols it won't work for others. From a user standpoint it's easier just to tell them to do the tuning for every single receiver instead of maintaining a list of which protocol could do it automatically or not with exceptions for some receivers which don't have telemetry or bound in a way that the telemetry is disabled.

**Reply by mha1** [reply]
Thank you very much for your answer. I understand your reasoning about user experience, especially in the case of protocols that have receivers with or without telemetry. For telemetry only protocols like HoTT I think it might be a tiny bit better to use the CC2500 generated offsets and to disable manual tuning in the header option byte.

I can think of two advantages of doing this:
1. Users may forget to do the manual tuning once in a while and are left with a less than perfect default of 0.
2. I presume the CC2500 offset value is a dynamically calculated value while operating that takes care of e.g. temperature drifts. So it is not a one-time generated static value and might better adjust to changing ambient conditions.

Thanks again for taking the time to answer my question!
